src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java

```diff
@@ -318,6 +319,29 @@ public boolean saveBatch(List<Pair<Indexable, Persistable>> models) throws Excep
         }
     }
 
+    public void deleteBatch(Collection<? extends Indexable> keys, Class<? extends Persistable> type)
+            throws RocksDBException {
+        ColumnFamilyHandle handle = classTreeMap.get(type);
+        ColumnFamilyHandle metadataHandle = metadataReference.get(type);
+
+        try (WriteBatch writeBatch = new WriteBatch()) {
+            for (Indexable key : CollectionUtils.emptyIfNull(keys)) {
+                byte[] keyBytes = key.bytes();
+                writeBatch.remove(handle, keyBytes);
+                if (metadataReference != null) {
+                    writeBatch.remove(metadataHandle, keyBytes);
+                }
+            }
+
+            WriteOptions writeOptions = new WriteOptions()
+                    //We are explicit about what happens if the node reboots before a flush to the db
+                    .setDisableWAL(false)
+                    //We want to make sure deleted data was indeed deleted
+                    .setSync(false);
```

Comment:
If you want to make sure that the write succeeded, don't you want to `setSync(true)`?

Reply:
yup...

Reply:
By the way in #924 I will check what is the cost of this